### PR TITLE
:sparkles: feat(types): Specification of domain types

### PR DESCRIPTION
**Pull request by ctacquet**
# Pull request for Candy-doc

## Your checklist for this pull request

- [X] Make sure you are requesting to **pull a feature/bugfix branch** (right
  side). Don't request your master!
- [X] Make sure you are making a pull request against the **main branch** (left
  side).
- [X] Check the commit's or even all commits' message styles matches our
  requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit
  test.

## Description

According to [DDD Distilled](https://sd.blackball.lv/library/Domain-Driven_Design_Distilled_(Vaughn_Vernon)_(2016).pdf#page=48) we can distinguish 3 primary types of subdomains.

This PR responds to #163 

❤️ Thank you!

### CODE REVIEW

**Comment by ctacquet** [comment]
Waiting for a first review from @ecattez

**Review by ecattez — CHANGES_REQUESTED** [review]

File: src/main/java/io/candydoc/plugin/model/BoundedContextDto.java
Comment (ecattez) [inline src/main/java/io/candydoc/plugin/model/BoundedContextDto.java:20]: Impossibe. It is a domain model. You have to create a dto or use String instead

**Review by ecattez — CHANGES_REQUESTED** [review]

File: src/main/java/io/candydoc/ddd/SubdomainType.java
Comment (gelhaimer) [inline src/main/java/io/candydoc/ddd/SubdomainType.java:3]: A subdomain can be seen as a domain. It only indicates that this domain exists within a bigger set.

I dont thing there is a need to have a full `@Subdomaintype` annotation.

It might only be an optional property of the `@Subdomain` annotation.

Having a `@SubdomainType` without `@Subdomain` on a class does not make much sense.

File: src/main/java/io/candydoc/ddd/SubdomainType.java
Comment (gelhaimer) [inline src/main/java/io/candydoc/ddd/SubdomainType.java:4]: Name of the subdomain type is inconsistent with other type.

Why is this one no called `CoreSubdomain` ?
Why do we need to suffix this although the type is about subdomains ?

It's like having an `Orientation` type and with `NorthOrientation`. `Orientation` doesn't provide any valuable information in `NorthOrientation` as it is already contextualised within the type.

**Review by gelhaimer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
